src/language-js/print/decorators.js

```diff
@@ -0,0 +1,73 @@
+"use strict";
+
+const { isNonEmptyArray } = require("../../common/util");
+const {
+  builders: { line, hardline },
+} = require("../../document");
+const { locStart } = require("../loc");
+const {
+  getParentExportDeclaration,
+  hasNewlineBetweenOrAfterDecorators,
+  isExportDeclaration,
+} = require("../utils");
+
+function printDecorators(path, options, print) {
+  const node = path.getValue();
+  const parentExportDecl = getParentExportDeclaration(path);
+  if (
+    isNonEmptyArray(node.decorators) &&
+    // If the parent node is an export declaration and the decorator
+    // was written before the export, the export will be responsible
+    // for printing the decorators.
+    !(
+      parentExportDecl &&
+      locStart(parentExportDecl, { ignoreDecorators: true }) >
+        locStart(node.decorators[0])
+    )
+  ) {
+    const shouldBreak =
+      node.type === "ClassExpression" ||
+      node.type === "ClassDeclaration" ||
+      hasNewlineBetweenOrAfterDecorators(node, options);
+    const separator = shouldBreak ? hardline : line;
+    const parts = [];
+    if (parentExportDecl) {
+      parts.push(hardline);
+    }
+
+    path.each((decoratorPath) => {
+      parts.push(print(decoratorPath, "expression"), separator);
+    }, "decorators");
+
+    return parts;
```

Comment:
```suggestion
    return [
      parentExportDecl ? hardline : shouldBreak ? breakParent : "",
      join(line, path.map(print, "decorators")),
      line,
    ];
```

Reply:
Printing `line` in `case "Decorator":` in `printer-estree.js` would be even better.